metabase/metabase
# Fix column scaling ("Multiply by a number") bugs

### Issues Fixed

- Closes: https://github.com/metabase/metabase/issues/12214
- Closes: https://github.com/metabase/metabase/issues/43536

---
### Manual Y axis min/max don't take the "Multiply by" factor into account (#12214)

**Demo**

https://github.com/metabase/metabase/assets/22608765/8e398917-d6e5-4b8a-b145-ef83e249c74b


**How to Verify**
1. Create a line chart with column scaling added to the metric/y-axis column
2. Turn on `"Linear"` y-axis scale
3. Turn off `"Auto y-axis range"
4. Input a custom range that is outside the bounds of the current min/max
   - customMin < minimum dataset value
   - customMax > maximum dataset value
5. See that the exact values you inputted are selected
6. Go back to step 1->6 and verify with:
   - Different y-axis scales: `["Log", "Pow"]`
   - Different chart types: `["Bar", "Area", "Combo", "Scatter", "Waterfall", "Stacked Bar", "Stacked Area"]`

**Acceptance Criteria**
- [ ] Column scaling/`multiply by` factor should still work as expected
- [ ] The custom y-axis range given should be used as is and should not have to be adjusted based on data transformations - y-axis scale type (linear, pow, log) and metric value scaling
  - [ ] This behavior should be extended to all relevant charts (line, area, bar, scatter, waterfall, stacked/non-stacked)
- [ ] Loki variant added to verify and enforce the expected behavior

---
### Using scaling on a question with multiple y-axis columns breaks tool-tips and y-axis label formatting (#43536)

**Demo**

https://github.com/metabase/metabase/assets/22608765/b396b5e7-f041-4902-af79-e5506e97d778

**How to Verify**
1. Create a line chart with multiple aggregations (`["Sum of Total", "Sum of Subtotal"]`) and one breakout (`"Created at: Year"`)
2. Toggle on data labels: `"Show values for data points"`
3. Set both metrics/y-axis columns selected to use the left y-axis.
4. Add column scaling to the either one of the metrics (but do not both with the same value)
5. See that the charts are plotted and behave as expected.
   - All of the data point values line up with the correct corresponding y-axis tick value
   - All of the data point values behave as expected: i.e. are displaying the correct column scaled value
6. Update the chart type to be stacked bar/stacked area
7. Turn the data labels on for the inside stacked segments as well (`"Both"`)
8. See that the charts inside labels and bar sizes relative to their values behave as expected.
9. Hover over one of the stacks and verify that the tool-tip contains the correct values for both series (correct values are the correctly scaled values based off of each column's own scale factor)

**Acceptance Criteria**
- [ ] Using column scaling with multiple y-axis metrics should generate the chart as expected
  - [ ] If one column uses scaling and the other does not, we should still see that each of the plotted data points line up with the correct corresponding y-axis tick value
- [ ] E2E test has been added to verify and enforce the expected behavior
